### PR TITLE
App component

### PR DESCRIPTION
**Pull request by Szulerinio**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -52,6 +43,14 @@ export class AppComponent implements OnInit {
       .subscribe((categories) => (this.categories = categories));
   }
 
+  public setNumberOfJokesToFetch(event: number): void {
```
Comment (ogigg) [inline src/app/app.component.ts:46]: `public setNumberOfJokesToFetch(numberOfJokesToFetch: number): void {`

File: src/app/app.component.ts [hunk 2]
```diff
@@ -76,4 +75,26 @@ export class AppComponent implements OnInit {
     console.log(event);
     this.getJoke();
   }
+  public saveJokes() {
+    forkJoin(
```
Comment (ogigg) [inline src/app/app.component.ts:79]: you don't need forkJoin here
```
this.jokeService.downloadJokes(
        this.numberOfJokesToFetch,
        this.name,
        this.chosenCategories
      ).subscribe({
      next(jokes) {
          ...
     })
```

File: src/app/app.component.ts [hunk 3]
```diff
@@ -76,4 +75,26 @@ export class AppComponent implements OnInit {
...
+      this.jokeService.downloadJokes(
+        this.numberOfJokesToFetch,
+        this.name,
+        this.chosenCategories
+      )
+    ).subscribe({
+      next(jokes) {
```
Comment (ogigg) [inline src/app/app.component.ts:86]: This works? I've never seen this syntax before. I was always using `next: (jokes) => { ... })`

File: src/app/app.component.ts [hunk 4]
```diff
@@ -76,4 +75,26 @@ export class AppComponent implements OnInit {
...
+        var element = document.createElement('a');
```
Comment (ogigg) [inline src/app/app.component.ts:87]: use angular's `Renderer2` to manipulate the DOM - https://www.digitalocean.com/community/tutorials/angular-using-renderer2

File: src/app/services/joke.service.ts [hunk 5]
```diff
@@ -48,8 +48,13 @@ export class JokeService {
...
-      environment.apiURL + Constants.API.CATEGORIES
-    );
+    return this.http.get<string[]>(environment.apiURL + API.CATEGORIES);
+  }
+
+  public downloadJokes(amount: number, name: string, categories: string[]) {
+    const observables: Observable<string>[] = new Array(amount)
+      .fill(0)
+      .map((_) => this.getJoke(name, categories));
+    return observables;
```
Comment (ogigg) [inline src/app/services/joke.service.ts:58]: consider adding forkJoin here

File: src/app/app.component.ts (hunk 2 above)
Comment (ogigg) [reply src/app/app.component.ts:79]: Oh I see, `downloadJokes` returns array of observables. In this case it's OK, but I think you can move `forkJoin` to `downloadJokes` method in joke.service so it's always returning just one ready to use observable

**Review by ogigg — APPROVED** [review]

File: src/app/app.component.ts (hunk 3 above)
Comment (Szulerinio) [reply src/app/app.component.ts:86]: Yes, it's [objects methods shorthand ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions#description)

being honest I didn't think of it, as docs are proposing your way of doing it. I only considered it when i saw that in my colleague's code

**Pull request merged**